doc/topics/blocking-vs-non-blocking.md

```diff
@@ -0,0 +1,143 @@
+# Overview of Blocking vs Non-Blocking
+
+This overview covers the difference between **blocking** and **non-blocking**
```

Comment:
Maybe we should clarify around **blocking** being synonymous with synchronous and **non-blocking** with asynchronous as all four terms a used throughout this document.